### PR TITLE
Git + GitHub additional skill

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: additional-skills/git/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Working with Remotes
+
+* Demonstrate adding a remote to an existing repository
+* Demonstrate pushing to a remote repository
+* Explain the difference between `git pull` and `git fetch`
+* Explain the difference between `origin/master` and `master`
+* Demonstrate setting the default upstream branch for a local branch
+* Demonstrate force pushing rebases into isolated upstream Pull Request branches
+
+## GitHub
```
Comment (craigjbass) [inline additional-skills/git/README.md]: I couldn't think of anything else of interest on here. Maybe commenting?

Comment (StevenLeighton21) [reply additional-skills/git/README.md]: `Demonstrate appropriate rebasing of isolated branches` assuming this means rebasing/squashing commit history?

Comment (craigjbass) [reply additional-skills/git/README.md]: yea

Comment (craigjbass) [reply additional-skills/git/README.md]: Don't merge a branch with 50 commits

Comment (StevenLeighton21) [reply additional-skills/git/README.md]: Words to live by.

Comment (StevenLeighton21) [reply additional-skills/git/README.md]: Perhaps something around PR etiquette, but I don't really have anything to add. LGTM!

**Comment by craigjbass** [comment]
Git ignore missing

**Comment by craigjbass** [comment]
Identifying files to ignore

**Comment by craigjbass** [comment]
Etiquette around getting reviews / useful practices around leaving reviews

**Comment by craigjbass** [comment]
Ensure all commits pass tests

**Comment by StevenLeighton21** [comment]
Seems more etiquette, not using git. Useful to know, but not core git?

File: additional-skills/git/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Git + GitHub
+
+All sections are required.
+
+## Initializing 
+
+* Demonstrate initializing a new git repository
+* Demonstrate creating an empty root commit
```
Comment (adrianclay) [inline additional-skills/git/README.md]: Had not thought about doing this, nice.

File: additional-skills/git/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+* Can explain the reasoning for creating an empty root commit
+
+## Basics
+
+* Demonstrate usage of `git status`
+* Demonstrate staging files for commit
+* Demonstrate committing staged files
+* Demonstrate committing a subset of changed files using staged and unstaged files
```
Comment (adrianclay) [inline additional-skills/git/README.md]: How about staging only a subset of a file?

File: additional-skills/git/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+* Demonstrate merging two refs to create a new ref
+* Demonstrate resolving merge conflicts
+
+## Rebasing
+
+* Demonstrate doing a simple rebase of master onto a branch 
+* Demonstrate performing an amend to previous commit
+* Demonstrate performing an interactive rebase of a branch
+  * Demonstrate using all possible rebase options
```
Comment (adrianclay) [inline additional-skills/git/README.md]: Not sure what I’d be assessing here.

**Comment by adrianclay** [comment]
Needs something about what a good commit message looks like.

File: additional-skills/git/README.md (hunk 3 above)
Comment (craigjbass) [reply additional-skills/git/README.md]: Yeah I don’t do this

**Comment by craigjbass** [comment]
Going to merge and address commit messages in upcoming

**Pull request merged**